.github/workflows/macos.yml

```diff
@@ -0,0 +1,263 @@
+      PYTHON_DIST_DIR: interface/python_binding/dist
+
+      VULKAN_SDK_VER: ${{ matrix.vk_sdk_ver }}
+    # This is just to tell the Vulkan install script where to install.
+    # NOTE: Hardcoded $HOME expansion, due to 'env' context being unavailble in env:
+      VULKAN_INSTALL_DIR: /Users/runner/VulkanSDK/${{ matrix.vk_sdk_ver }}
+      VULKAN_SDK: /Users/runner/VulkanSDK/${{ matrix.vk_sdk_ver }}/macOS
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          # Fetch all history to make sure tags are
+          # included (used for version creation)
+          fetch-depth: 0
+
+      - name: before_install
+        run: |
+          echo -n "Running on the following GitHub Actions CI runner: " && uname -a
+          echo -n "CMake version on the runner is " && cmake --version
+          echo -e "machine github.com\n  login $GITHUB_TOKEN" >> ~/.netrc # Prevent rate limiting on Git LFS.
+          cat ~/.netrc
+          if [[ -n "$FEATURE_LOADTESTS" && "$FEATURE_LOADTESTS" != "OFF" ]]; then
+            # Install vcpkg.
+            pushd $HOME
+            git clone https://github.com/microsoft/vcpkg
+            cd vcpkg && ./bootstrap-vcpkg.sh -disableMetrics && echo "VCPKG_ROOT=`pwd`" >> $GITHUB_ENV
+            popd
+          fi
+
+      - name: Install Doxygen
+        if: matrix.options.doc
+        uses: ssciwr/doxygen-install@v1
+
+      - name: Install Graphviz
+        if: matrix.options.doc
+        uses: ts-graphviz/setup-graphviz@v2
+
+      - name: Run install script
+        run: ./scripts/install_macos.sh
+
+      - name: Set up Python 3.11
+        if: matrix.options.py == 'ON'
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11.4'
+
+      - name: before_script
+        run: |
+          ./scripts/before_build_macos.sh
+          if [ "$FEATURE_TOOLS_CTS" = "ON" ]; then
+            git submodule update --init --recursive tests/cts
+          fi
+          # Make sure embedded dates are correct.
+          ./install-gitconfig.sh
+          scripts/smudge_date.sh
+
+      - name: script
+        run: |
+          set -e # Instead of trying to chain everything together with &&
+                 # otherwise subsequent commands will swallow bad exit status.
+          if [ "$PLATFORM" = "macOS" ]; then
+            ./scripts/build_macos.sh
+          else
+            ./scripts/build_ios.sh
+          fi
+          if [ "$DEPLOY_DOCS" = "YES" ]; then
+            # Some files in the pyktx docs have an _ prefix so Jekyll will
+            # not copy them from the gh-pages branch to the website. This
+            # file says no Jekyll files here. Treat all as ordinary files.
+            touch $BUILD_DIR/docs/html/.nojekyll
+          fi
+          set +e
+
+      - name: Notarize
+        if: matrix.options.platform == 'macOS'
+        run: |
+          ./scripts/notarize.sh $BUILD_DIR/KTX-Software-*.pkg $APPLE_ID $DEVELOPMENT_TEAM $ALTOOL_PASSWORD && export NOTARIZED="true"
```

Comment:
Notarization is failing unnoticed in the test builds you did. The reason for the failure is that I changed the team environment variable name to `APPLE_DEVELOPMENT_TEAM` to disambiguate because the secrets list now has Windows and Apple stuff in it.

I'm not sure why the failure was unnoticed. I thought it might be because it was called in an `if`, `else` block but you have `set -e` so I'm not sure.

Reply:
Just looked at the latest log. The failure is being noticed now it is a separate step.